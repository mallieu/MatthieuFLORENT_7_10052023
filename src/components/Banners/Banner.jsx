const Banner = ({ className, title, HasBackgroundImage }) => (
    <div
        className={className}
        style={
            HasBackgroundImage
                ? { backgroundImage: `url(${HasBackgroundImage})` }
                : null
        }
    >
        {title ? <h1>{title}</h1> : null}
    </div>
);

export default Banner;
