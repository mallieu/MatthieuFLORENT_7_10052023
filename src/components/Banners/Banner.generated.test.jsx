import renderer from 'react-test-renderer';
import Banner from './Banner';

const renderTree = (tree) => renderer.create(tree);
describe('<Banner>', () => {
    it('Banner - should render component <Banner />', () => {
        expect(renderTree(<Banner />).toJSON()).toMatchSnapshot();
    });
    it('Banner - should render component with props', () => {
        expect(
            renderTree(
                <Banner
                    className={'banner'}
                    title={'Je suis un test'}
                    HasBackgroundImage={'image.jpg'}
                />
            ).toJSON()
        ).toMatchSnapshot();
    });
});
