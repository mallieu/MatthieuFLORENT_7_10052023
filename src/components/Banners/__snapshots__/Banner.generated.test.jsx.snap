// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner> Banner - should render component <Banner /> 1`] = `
<div
  style={null}
/>
`;

exports[`<Banner> Banner - should render component with props 1`] = `
<div
  className="banner"
  style={
    Object {
      "backgroundImage": "url(image.jpg)",
    }
  }
>
  <h1>
    Je suis un test
  </h1>
</div>
`;
